src/Microsoft.AspNetCore.Blazor.Server/Circuits/CircuitHost.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Blazor.Browser.Rendering;
+using Microsoft.AspNetCore.Blazor.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Blazor.Server.Circuits
+{
+    internal class CircuitHost : IDisposable
+    {
+        public event UnhandledExceptionEventHandler UnhandledException;
```

Comment:
Do you know if there's any way we can wire this up to the SignalR hub so that, if there's an unhandled exception, we (1) send that info to the client so it can be displayed in the browser console and (2) terminate the SignalR connection?

Reply:
That's the plan. As stated, I've made no effort to do that.

Reply:
I've written what I think the error handling code will look like, but it is not doing what I expect. I'll get some advice about this tomorrow.